### ISSUE TITLE
[docs] Detail row panels are ignored by VoiceOver

### ISSUE THREAD

**Issue by mj12albert** [mui/mui-x#18455]
### Search keywords

voiceover master detail row panels

### Related page

https://mui.com/x/react-data-grid/master-detail/#what-is-the-master-detail-pattern

### Kind of issue

Broken demo

### Issue description

After expanding an expandable row, VoiceOver is unable to move into the expanded details row.

In the recording I expand row 1 (`Matheus`) at 00:18s and try to use VO+Left/Right/Up/Down arrows to move to details, but keep getting moved to the `Olivier` row

https://github.com/user-attachments/assets/55c67ca0-f82c-46c3-b134-fc30c84565a8

This was recorded on Safari 18.5 (not the latest) and MacOS 14.7 (not the latest); I see the same results on latest Chrome.

(Not sure if it's a bug or just docs, I didn't test other demos on the page)

### Context

Background: https://github.com/mui/material-ui/issues/46381#issuecomment-2990220380

**Comment by mj12albert** [comment]
More background: somebody in the Discord shared Tanstack table's expandable rows demo: https://codesandbox.io/p/sandbox/github/tanstack/table/tree/main/examples/react/expanding?file=%2Fsrc%2Fmain.tsx&from-embed=

I encounter a few issues right off the bat, but it does not skip any rows, even the crazy deep nested ones
